src/backend/drm/backend.rs

```diff
@@ -0,0 +1,503 @@
+                        },
+                        |surface| {
+                            // create an egl surface from the gbm one
+                            debug!(log, "Creating EGLSurface");
+                            let egl_surface = context.egl.create_surface(&surface)?;
+
+                            // make it active for the first `crtc::set`
+                            // (which is needed before the first page_flip)
+                            unsafe { egl_surface.make_current()? };
+                            egl_surface.swap_buffers()?;
+
+                            // init the first screen
+                            // (must be done before calling page_flip for the first time)
+                            let mut front_bo = surface.lock_front_buffer()?;
+                            debug!(log, "FrontBuffer color format: {:?}", front_bo.format());
+                            // we need a framebuffer per front buffer
+                            let fb = framebuffer::create(context.devices.drm, &*front_bo)?;
+
+                            debug!(log, "Initialize screen");
+                            crtc::set(
+                                context.devices.drm,
+                                crtc,
+                                fb.handle(),
+                                &connectors,
+                                (0, 0),
+                                Some(mode),
+                            )?;
+                            front_bo.set_userdata(fb);
+
+                            Ok(EGL {
+                                surface: egl_surface,
+                                buffers: GbmBuffers {
+                                    front_buffer: Cell::new(front_bo),
+                                    next_buffer: Cell::new(None),
+                                },
+                            })
+                        },
+                    ).map_err(DrmError::from)?,
+                })
+            })?,
+            crtc,
+            mode,
+            connectors,
+            own_id: Id(own_id),
+            logger: log.clone(),
+        })
+    }
+
+    pub(crate) fn unlock_buffer(&self) {
+        // after the page swap is finished we need to release the rendered buffer.
+        // this is called from the PageFlipHandler
+        self.graphics.rent(|gbm| {
+            gbm.surface.rent(|egl| {
+                let next_bo = egl.buffers.next_buffer.replace(None);
+
+                if let Some(next_buffer) = next_bo {
+                    trace!(self.logger, "Releasing all front buffer");
+                    egl.buffers.front_buffer.set(next_buffer);
+                // drop and release the old buffer
+                } else {
+                    unreachable!();
+                }
+            })
+        });
+    }
+
+    pub(crate) fn is_crtc(&self, crtc: crtc::Handle) -> bool {
+        crtc == self.crtc
+    }
+}
+
+impl DrmBackend {
+    /// Add a connector to backend
```

Comment:
It wold be good to have either a basic explanation of what a "connector" is, or at least a link to the relevant documentation of the drm crate.

I for one I absolutely no idea of what it is. :)

Reply:
Noted. A "connector" is just what it sounds like. A port on your computer possibly with a connected monitor.